cmd/root.go

```diff
@@ -97,18 +100,30 @@ func startGohaqd(cmd *cobra.Command, args []string) {
 		QueueUrl:        q.QueueUrl,
 		WaitTimeSeconds: aws.Int64(20),
 	}
+	sem = make(chan *sqs.Message)
+
+	for i := 0; i < parallelRequests; i++ {
+		go startConsumer(q.QueueUrl)
+	}
+
 	for {
-		pollSQS(q.QueueUrl)
+		pollSQS()
 	}
 }
 
-func pollSQS(queueURL *string) {
+func pollSQS() {
 	resp, err := svc.ReceiveMessage(msgparams)
 	if err != nil {
 		log.Fatalf(err.Error())
 	}
 
 	for _, msg := range resp.Messages {
+		sem <- msg
```

Comment:
@ApsOps Would this be blocking if sem already has a message?
Buffered channel should eliminate direct dependency between `pollSqs` & `startConsumer`.